### DISCUSSION TITLE
Unable to dismiss Notify when page route changes

### DISCUSSION THREAD

**Discussion by cdecinkoKnight** [quasarframework/quasar#15689] [General - Components / Directives / etc] [answered]
I have a page that displays an error Notify when there are no records found. It hangs on the screen for a few seconds then goes away. If the user navigates back to the home page before the Notify times out, it's still there saying no records which isn't true anymore.

I have an error object that I populate with the message, color and position and pass it to $q.notify via a watcher. When the values change, show the Notify. Otherwise when the values are cleared, I'd like to remove the notify immediately.

If I create an empty notify outside my if/else I get an empty notify showing up when it should not. Right now, I can create the variable outside the if/else and check for an error, then display. But trying to dismiss the notify won't work because I get "Cannot invoke an object which is possibly 'undefined'".

`    

    const dataErrors = dataRequestError.value;  
    watch(dataErrors, (error) => {  
      let errorMsg;  
      if (error.error) {  
        showTableData.value = true;
        $q.loading.hide();
        $q.notify;
        errorMsg = $q.notify({
          message: error.errorMessage,
          color: "red",
          position: "top",
        });
      } else {
        errorMsg();
      }
    });
`

**Comment by rstoenescu** [comment, marked as answer]
https://quasar.dev/quasar-plugins/notify#programmatically-closing

Use the programmatic closing and watch the route yourself.

```
import { useRoute } from 'vue-router'
import { watch } from 'vue'

const $route = useRoute()
const dismiss = $q.notify({...})

watch(() => $route.path, dismiss)
```
